quickjs.c

```diff
@@ -32773,8 +32773,10 @@ static JSValue js_create_function(JSContext *ctx, JSFunctionDef *fd)
             }
         } else {
             b->vardefs = (void *)((uint8_t*)b + vardefs_offset);
-            memcpy(b->vardefs, fd->args, fd->arg_count * sizeof(fd->args[0]));
-            memcpy(b->vardefs + fd->arg_count, fd->vars, fd->var_count * sizeof(fd->vars[0]));
+            if (fd->vars && fd->arg_count > 0) {
```

Comment:
Can `fd->vars` be `NULL`?

Reply:
ive rebase and squashed. yes, fd->vars was null in one of the tests i was running, thats why i add the check, but if you want i can try again witohut this condition and try to repro and share the reproducer.